jerry-core/ecma/base/ecma-helpers-string.cpp

```diff
@@ -1293,75 +1329,55 @@ ecma_compare_ecma_strings_relational (const ecma_string_t *string1_p, /**< ecma-
-    JERRY_ASSERT (RCS_RECORD_IS_CHARSET (lit));
-    return lit_charset_literal_get_length (lit);
-  }
-  else if (container == ECMA_STRING_CONTAINER_MAGIC_STRING)
-  {
-    TODO ("Cache magic string lengths")
-    return lit_utf8_string_length (lit_get_magic_string_utf8 (string_p->u.magic_string_id),
-                                   lit_get_magic_string_size (string_p->u.magic_string_id));
-  }
-  else if (container == ECMA_STRING_CONTAINER_MAGIC_STRING_EX)
-  {
-    TODO ("Cache magic string lengths")
-    return lit_utf8_string_length (lit_get_magic_string_ex_utf8 (string_p->u.magic_string_ex_id),
-                                   lit_get_magic_string_ex_size (string_p->u.magic_string_ex_id));
-  }
-  else if (container == ECMA_STRING_CONTAINER_UINT32_IN_DESC)
+  switch ((ecma_string_container_t) string_p->container)
   {
-    const uint32_t uint32_number = string_p->u.uint32_number;
-    const uint32_t max_uint32_len = 10;
-    const uint32_t nums_with_ascending_length[10] =
+    case ECMA_STRING_CONTAINER_LIT_TABLE:
     {
-      1u,
-      10u,
-      100u,
-      1000u,
-      10000u,
-      100000u,
-      1000000u,
-      10000000u,
-      100000000u,
-      1000000000u
-    };
-
-    ecma_length_t length = 1;
-
-    while (length < max_uint32_len
-           && uint32_number >= nums_with_ascending_length[length])
+      lit_literal_t lit = lit_get_literal_by_cp (string_p->u.lit_cp);
+      JERRY_ASSERT (RCS_RECORD_IS_CHARSET (lit));
+      return lit_charset_literal_get_length (lit);
+    }
+    case ECMA_STRING_CONTAINER_MAGIC_STRING:
     {
-      length++;
+      TODO ("Cache magic string lengths")
+      return lit_utf8_string_length (lit_get_magic_string_utf8 (string_p->u.magic_string_id),
+                                     lit_get_magic_string_size (string_p->u.magic_string_id));
     }
+    case ECMA_STRING_CONTAINER_MAGIC_STRING_EX:
+    {
+      TODO ("Cache magic string lengths")
+      return lit_utf8_string_length (lit_get_magic_string_ex_utf8 (string_p->u.magic_string_ex_id),
+                                     lit_get_magic_string_ex_size (string_p->u.magic_string_ex_id));
+    }
+    case ECMA_STRING_CONTAINER_UINT32_IN_DESC:
+    {
+      const uint32_t uint32_number = string_p->u.uint32_number;
+      ecma_length_t length = 1;
 
-    return length;
-  }
-  else if (container == ECMA_STRING_CONTAINER_HEAP_NUMBER)
-  {
-    const ecma_number_t *num_p = ECMA_GET_NON_NULL_POINTER (ecma_number_t,
-                                                            string_p->u.number_cp);
-
-    lit_utf8_byte_t buffer[ECMA_MAX_CHARS_IN_STRINGIFIED_NUMBER];
+      while (length < max_uint32_len
+             && uint32_number >= nums_with_ascending_length[length])
+      {
+        length++;
+      }
 
-    return (ecma_length_t) ecma_number_to_utf8_string (*num_p, buffer, sizeof (buffer));
-  }
-  else
-  {
-    JERRY_ASSERT (container == ECMA_STRING_CONTAINER_HEAP_CHUNKS);
+      return length;
```

Comment:
This `ECMA_STRING_CONTAINER_UINT32_IN_DESC` case is still here, a clone of the same case in `ecma_string_get_size`, and not factored out into a helper. Intentional?